LDAP/Init.hsc

```diff
@@ -106,6 +107,17 @@ ldapSimpleBind ld dn passwd =
            return ()
                          )))
 
+{- | Bind with the SASL EXTERNAL mechanism. -}
+ldapExternalSaslBind :: LDAP   -- ^ LDAP Object
+                     -> String -- ^ Authorization identity (UTF-8 encoded; pass "" to derive it from the authentication identity)
```

Comment:
Yes, it should be a `Bytestring`, just like most other `String` values in this package (starting with attribute values). I didn't want to diverge from the established convention for this function, though I'd gladly switch to more appropriate types throughout the package in a separate move. Not sure how to do that gracefully, though.